gobblin-cluster/src/main/java/org/apache/gobblin/cluster/SleepingTask.java

```diff
@@ -64,19 +70,27 @@ public void run() {
         throw new IOException("File creation error: " + taskStateFile.getName());
       }
       long endTime = System.currentTimeMillis() + sleepTime * 1000;
-      while (System.currentTimeMillis() <= endTime) {
-        Thread.sleep(1000L);
-        log.warn("Sleeping for {} seconds", sleepTime);
-      }
+
+      AssertWithBackoff.create().maxSleepMs(1000).timeoutMs(Integer.MAX_VALUE).backoffFactor(1).
+          assertTrue(new Predicate<Void>() {
+            @Override
+            public boolean apply(@Nullable Void input) {
+              return System.currentTimeMillis() > endTime;
+            }
+          }, "Waiting for the job to start...");
+
       log.info("Hello World!");
       super.run();
     } catch (InterruptedException e) {
-      log.error("Sleep interrupted.");
+      log.info("Sleep interrupted.");
```

Comment:
Nit: Prefer to keep the log level here to error, given that this is an exception we are propagating.

Reply:
I think I wrongly wrote it error. warn seems more appropriate to me. If it comes here, it does not mean that the SleepingTask did anything **erroneous**, but got interrupted. could be info too, but looks like an important message, especially in SleepingTask